Wikipedia/Code/NotificationsCenterViewController.swift

```diff
@@ -70,17 +77,35 @@ final class NotificationsCenterViewController: ViewController {
         enableToolbar()
         setToolbarHidden(false, animated: false)
 
-		let editButton = UIBarButtonItem(title: WMFLocalizedString("notifications-center-edit-button", value: "Edit", comment: "Title for navigation bar button to toggle mode for editing notification read status"), style: .plain, target: self, action: #selector(userDidTapEditButton))
-		navigationItem.rightBarButtonItem = editButton
-	}
-
-	// MARK: - Edit button
-
-	@objc func userDidTapEditButton() {
-
+		navigationItem.rightBarButtonItem = editButtonItem
+        isEditing = false
 	}
 
 	// MARK: - Public
+    
+    override func setEditing(_ editing: Bool, animated: Bool) {
+        super.setEditing(editing, animated: animated)
+        
+        notificationsView.collectionView.allowsSelection = editing
+        notificationsView.collectionView.allowsMultipleSelection = editing
+        
+        //Reconfigure visible cells to reflect new edit mode.
+        //This has a smoother change vs reloadData()
+        if #available(iOS 15.0, *) {
+            if var snapshot = dataSource?.snapshot() {
+                snapshot.reconfigureItems(snapshot.itemIdentifiers)
+                dataSource?.apply(snapshot)
```

Comment:
In this case, I think it may be preferable to not animate: `dataSource?.apply(snapshot, animatingDifferences: false)` – it's unexpected seeing the leading image fade between states rather than just discretely change between states.